device-server/src/main/kotlin/com/badoo/automation/deviceserver/ios/device/Device.kt

```diff
@@ -366,9 +366,26 @@ class Device(
         remote.fbsimctl.uninstallApp(udid, bundleId)
     }
 
+    fun runXcuiTest(xcuiTestExecutionConfig: XcuiTestExecutionConfig): Map<String, String> {
+        logger.debug(logMarker, "Running XCUI test '${xcuiTestExecutionConfig.testName}' of '${xcuiTestExecutionConfig.appName}'" +
+                " using '${xcuiTestExecutionConfig.pathToDirWithXctestrunFile}/${xcuiTestExecutionConfig.xctestrunFileName}' on device $this")
+
+        val command = listOf("xcodebuild", "test-without-building", "-xctestrun",
+                "${xcuiTestExecutionConfig.pathToDirWithXctestrunFile}/${xcuiTestExecutionConfig.xctestrunFileName}",
```

Comment:
Do I understand correctly, that this will work on localhost only, as it assumes that files for testing are available on the node host?
What if I want to run tests on a remote node (ssh), do I need to copy files manually beforehand?

Reply:
Yes, this will work only if files for testing are available on the node host. And for now you should manually copy files before run tests.

So, should ios-device-server be able to transfer files for testing to all needed nodes by itself (by rsync, for example)? Or lets leave it for tests-runners or third-party utilities?